crates/ol/stf/src/update.rs

```diff
@@ -0,0 +1,89 @@
+use strata_acct_types::{AccountId, AcctError, BitcoinAmount, MsgPayload};
+use strata_ledger_types::{AccountTypeState, Coin, IAccountState, IL1ViewState, StateAccessor};
+use strata_snark_acct_sys::{handle_snark_msg, handle_snark_transfer};
+
+use crate::{
+    context::BlockExecContext,
+    error::StfResult,
+    system_handlers::{get_system_msg_handler, get_system_transfer_handler},
+};
+
+/// Sends a message with attached value from one account to another.
+///
+/// Creates a [`Coin`] for the message value and adds it to the recipient's balance.
+/// Routes to system handlers for system accounts, or to account-type-specific handlers.
+///
+/// # Safety
+/// Creates a coin with `new_unchecked` - caller must ensure sender has sufficient balance.
+/// The recipient's `add_balance` implementation must call `safely_consume_unchecked` on the coin.
+pub(crate) fn send_message<S: StateAccessor>(
+    ctx: &BlockExecContext,
+    state_accessor: &mut S,
+    from: AccountId,
+    to: AccountId,
+    msg_payload: &MsgPayload,
+) -> StfResult<()> {
+    let cur_epoch = state_accessor.l1_view().cur_epoch();
+    let Some(target_acct) = state_accessor.get_account_state_mut(to)? else {
+        return Err(AcctError::NonExistentAccount(to).into());
+    };
+
+    // First update the balance
```

Comment:
This function is called `send_message`, what exactly are we doing here? who updates the balance? Write explicitly what balance we are updating (seems to be the target). Who is the sender and how can it update the target's balance?